source/interprocedural_analyses/taint/modelParser.ml

```diff
@@ -1124,19 +1120,13 @@ let parse_find_clause ~path ({ Node.value; location } as expression) =
       | "methods" -> Ok ModelQuery.MethodModel
       | "attributes" -> Ok ModelQuery.AttributeModel
       | unsupported ->
-          Error
-            (invalid_model_error
-               ~path
-               ~location
-               ~name:"model query"
-               (Format.sprintf "Unsupported find clause `%s`" unsupported)))
+          Error (model_verification_error ~path ~location (UnsupportedFindClause unsupported)))
   | _ ->
       Error
-        (invalid_model_error
+        (model_verification_error
            ~path
            ~location
-           ~name:"model query"
-           (Format.sprintf "Find clauses must be strings, got: `%s`" (Expression.show expression)))
+           (InvalidFindClauseType (Expression.show expression)))
```

Comment:
Here and other places: could we just store the expression directly in the error? We already do that for `InvalidIdentifier` for instance.